src/TakeoffController.cpp

```diff
@@ -160,25 +162,51 @@ bool TakeoffController::update(const ros::Time& time,
             ROS_ERROR("Arming service failed");
             return false;
         }
-        state_ = TakeoffState::RAMP;
+        if (!calibrateThrustModel(time)) {
+            ROS_ERROR("Failed to calibrate thrust model");
+            return false;
+        }
+        arm_time_=time;
+        state_ = TakeoffState::PAUSE;
     }
-    else if(state_ == TakeoffState::RAMP) {
-        if(!landing_detected_message_.data) {
+    else if (state_ == TakeoffState::PAUSE){
+        if (time > arm_time_ + post_arm_delay_){
+            state_ = TakeoffState::RAMP;
             if (!calibrateThrustModel(time)) {
                 ROS_ERROR("Failed to calibrate thrust model");
                 return false;
             }
-          state_ = TakeoffState::DONE;
+            ramp_start_time_ = time;
         }
-        // Check if UAV is above switch sensing height, if it is go to safety response
-        else if(transform.transform.translation.z > takeoff_max_height_switch_pressed_)
-        {
-            ROS_ERROR("Takeoff handler failed, quad's switches are toggled, but quad is above toggle height");
-            return false;
+    }
+    else if(state_ == TakeoffState::RAMP) {
+        if (time <= ramp_start_time_ + takeoff_throttle_ramp_duration_){
+            double voltage;
+            if (!battery_interpolator_.getInterpolatedMsgAtTime(voltage, time)) {
+                ROS_ERROR("Failed to get battery voltage to calibrate thrust model");
+                return false;
+            }
+
+            geometry_msgs::TransformStamped transform;
+            bool success = transform_wrapper_.getTransformAtTime(transform,
+                                                                 "map",
+                                                                 "center_of_lift",
+                                                                 time,
+                                                                 update_timeout_);
+            if (!success) {
+                ROS_ERROR("Failed to get current transform to calibrate thrust model");
+                return false;
+            }
+
+            geometry_msgs::PointStamped col_point;
+            tf2::doTransform(col_point, col_point, transform);
+
+            double hover_throttle = thrust_model_.throttleFromAccel(9.8, voltage, col_point.point.z);
```

Comment:
is the 9.8 the acceleration due to gravity? is this how we want to do this? if so, wouldn't it be better to provide more precision (i.e. 9.81 m/s^2)

Reply:
Actually, we have to figure out what g is equal to at the Georgia Tech venue and at Pitt and make that a param that goes to the lowest digit possible